force_wfmanager/central_pane/plot.py

```diff
@@ -174,8 +182,46 @@ def _update_plot_data(self):
         x_index = self.analysis_model.value_names.index(self.x)
         y_index = self.analysis_model.value_names.index(self.y)
 
-        self._plot_data.set_data('x', self._data_arrays[x_index])
-        self._plot_data.set_data('y', self._data_arrays[y_index])
+        if self._data_arrays != []:
+            self._plot_data.set_data('x', self._data_arrays[x_index])
+            self._plot_data.set_data('y', self._data_arrays[y_index])
+
+            self.resize_plot()
+
+    def resize_plot(self):
+        """ Sets the size of the current plot to have some spacing between the
+        largest/smallest value and the plot edge. Also returns the new values
+        (X min, X max, Y min, Y max) if the plot area changes or None if it
+        does not.
+        """
+        if self.x is None or self.y is None:
+            return None
+
+        x_data = self._plot_data.get_data('x')
+        y_data = self._plot_data.get_data('y')
+        if x_data is None:
+            return None
+
+        if len(x_data) > 1:
+            x_max = max(x_data)
+            x_min = min(x_data)
+            x_size = abs(x_max - x_min)
+            x_max = x_max + 0.1 * x_size
+            x_min = x_min - 0.1 * x_size
+
+            y_max = max(y_data)
+            y_min = min(y_data)
+            y_size = abs(y_max - y_min)
+            y_max = y_max + 0.1 * abs(y_size)
+            y_min = y_min - 0.1 * abs(y_size)
+
+            self._plot.range2d.x_range.low = x_min
+            self._plot.range2d.x_range.high = x_max
+            self._plot.range2d.y_range.low = y_min
+            self._plot.range2d.y_range.high = y_max
+
+            return x_min, x_max, y_min, y_max
```

Comment:
This return is pretty much just for information as the actual assignment has already taken place. I don't know if there are any conventions we follow about not mixing return lengths or anything like that